library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -173,6 +201,60 @@ class Client() {
         }
     }
 
+    private suspend fun ffiXmtpClient(
+        options: ClientOptions?,
+        account: SigningKey,
+        appContext: Context?,
+        privateKeyBundleV1: PrivateKeyBundleV1,
+    ): FfiXmtpClient? {
+        val libXMTPClient: FfiXmtpClient? =
+            if (options != null && options.enableLibXmtpV3 && options.appContext != null) {
+                val alias = "xmtp-${options.api.env}-${account.getAddress()}"
+
+                val dbDir = File(appContext?.filesDir?.absolutePath, "xmtp_db")
+                dbDir.mkdir()
+                val dbPath: String = dbDir.absolutePath + "/$alias.db3"
+
+                val sharedPreferences =
+                    options.appContext.getSharedPreferences("XMTPPreferences", Context.MODE_PRIVATE)
+                val dbEncryptionKey = sharedPreferences.getString(alias, null)
```

Comment:
In the long term, is it possible to put this in the Android Keystore? Shared preferences doesn't appear to be secure from people with access to the device, therefore the encryption key wouldn't provide any additional protection compared to leaving the database unencrypted

Reply:
So I went down that route here https://github.com/xmtp/xmtp-android/pull/156/commits/1e84e4c34abddd8e9bb4204bee46770d3f2c163e but I wasn't able to make it work as the keystore stuff didn't come till API 23 or 24. I'm going to bump our min sdk version to 23 and see if I can make it work.

Reply:
Seems like bumping to version 23 allowed me to use the keystore!